src/ScriptBuilder/Saga/Writers/OracleSagaScriptWriter.cs

```diff
@@ -0,0 +1,212 @@
+    {
+        var columnName = OracleCorrelationPropertyName(correlationProperty);
+        var indexName = CreateSagaIndexName(tableName, correlationProperty.Name);
+
+        writer.Write($@"
+select count(*) into n from user_indexes where table_name = '{tableName}' and index_name = '{indexName}';
+if(n = 0)
+then
+  sqlStatement := 'create unique index {indexName} on {tableName} ({columnName} ASC)';
+
+  execute immediate sqlStatement;
+end if;
+");
+    }
+
+    public void WritePurgeObsoleteProperties()
+    {
+        var builder = new StringBuilder();
+
+        if (saga.CorrelationProperty != null)
+        {
+            builder.Append($" and\r\n        column_name <> '{OracleCorrelationPropertyName(saga.CorrelationProperty)}'");
+        }
+        if (saga.TransitionalCorrelationProperty != null)
+        {
+            builder.Append($" and\r\n        column_name <> '{OracleCorrelationPropertyName(saga.TransitionalCorrelationProperty)}'");
+        }
+        writer.Write($@"
+select count(*) into n
+from all_tab_columns
+where table_name = '{tableName}' and column_name like 'CORR_%'{builder};
+
+if(n > 0)
+then
+
+  select 'alter table {tableName} drop column ' || column_name into sqlStatement
+  from all_tab_columns
+  where table_name = '{tableName}' and column_name like 'CORR_%'{builder};
+
+  execute immediate sqlStatement;
+
+end if;
+");
+    }
+
+    public void WritePurgeObsoleteIndex()
+    {
+        // Dropping the column with the index attached removes the index as well
+    }
+
+    public void WriteCreateTable()
+    {
+        writer.Write($@"
+  select count(*) into n from user_tables where table_name = '{tableName}';
+  if(n = 0)
+  then
+    
+    sqlStatement :=
+       'create table {tableName} 
+       (
+          id varchar2(38) not null,
+          metadata clob not null,
+          data clob not null,
+          persistenceversion varchar2(23) not null,
+          sagatypeversion varchar2(23) not null,
+          concurrency number(9) not null,
+          constraint {tableName}_PK primary key
+          (
+            id
+          )
+          enable
+        )';
+
+    execute immediate sqlStatement;
+
+  end if;
+");
+    }
+
+    public void WriteDropTable()
+    {
+        writer.Write($@"
+declare
+  n number(10);
+begin
+  select count(*) into n from user_tables where table_name = '{tableName}';
+  if(n > 0)
+  then
+    execute immediate 'drop table {tableName}';
+  end if;
+end;
+");
+    }
+
+    string OracleCorrelationPropertyName(CorrelationProperty property)
+    {
+        var name = "CORR_" + property.Name.ToUpper();
+        if (name.Length > 30)
+        {
+            name = name.Substring(0, 30);
+        }
+        return name;
+    }
+
+    // Generates a 30ch index name like "SAGAIDX_66462BF46C9DCB70FD257D" based on
+    // SHA1 hash of saga name and correlation property name
+    string CreateSagaIndexName(string sagaName, string correlationPropertyName)
```

Comment:
Perhaps I lack historical insight and context here but what are the problems? I know we use hashing in a couple of places (including NH persister ATTs) but I am not aware of problems this causes.

Limit is 64 for MySQL and 128 for SQL Server. Theoretically it could be a problem on these RDBMSs as well.

We could consider creating an API to control this along with guidance in docs but I'd much rather do this as a separate issue.

Thoughts?